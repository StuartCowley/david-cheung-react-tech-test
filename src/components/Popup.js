import React from "react";
import PropTypes from "prop-types";
import "../styles/popup.css";

const Popup = ({ setButtonPopup, src }) => {
  return (
    <div className="popup">
      <div className="popup-inner">
        <button className="close-btn" onClick={() => setButtonPopup(false)}>
          X
        </button>
        <img
          className="popup-href"
          src={src.href}
          alt={src.title}
          width="100%"
        />
        <p className="popup-title">{src.title}</p>
      </div>
    </div>
  );
};

export default Popup;

Popup.propTypes = {
  setButtonPopup: PropTypes.func.isRequired,
  src: PropTypes.shape({
    href: PropTypes.string,
    title: PropTypes.string,
  }).isRequired,
};
