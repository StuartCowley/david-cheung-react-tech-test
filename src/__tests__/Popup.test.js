import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Popup from "../components/Popup";

describe("Popup", () => {
  const validProps = {
    setButtonPopup: jest.fn(),
    src: { href: "test-src", title: "test-title" },
  };

  test("Renders as expected", () => {
    const rendered = renderer.create(
      <Popup setButtonPopup={validProps.setButtonPopup} src={validProps.src} />
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert one close button is present", () => {
    render(
      <Popup setButtonPopup={validProps.setButtonPopup} src={validProps.src} />
    );
    const buttons = screen.getAllByRole("button");

    expect(buttons).toHaveLength(1);
    expect(buttons[0]).toHaveTextContent("X");
  });

  test("Assert one img is present", () => {
    render(
      <Popup setButtonPopup={validProps.setButtonPopup} src={validProps.src} />
    );
    const imgs = screen.getAllByRole("img");

    expect(imgs).toHaveLength(1);
    expect(imgs[0]).toHaveAttribute("src", "test-src");
  });

  test("Assert one text element is present", () => {
    render(
      <Popup setButtonPopup={validProps.setButtonPopup} src={validProps.src} />
    );

    expect(screen.getByText("test-title")).toBeTruthy();
    expect(screen.getByText("test-title")).toBeInstanceOf(HTMLParagraphElement);
  });
});
