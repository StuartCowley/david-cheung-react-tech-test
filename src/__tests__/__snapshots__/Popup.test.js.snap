// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Renders as expected 1`] = `
<div
  className="popup"
>
  <div
    className="popup-inner"
  >
    <button
      className="close-btn"
      onClick={[Function]}
    >
      X
    </button>
    <img
      alt="test-title"
      className="popup-href"
      src="test-src"
      width="100%"
    />
    <p
      className="popup-title"
    >
      test-title
    </p>
  </div>
</div>
`;
